### DISCUSSION TITLE
ConfigDB().get() with default values

### DISCUSSION THREAD

**Discussion by timothyscherer** [pyuvm/pyuvm#177] [Ideas]
I am trying to create a test bench architecture that fully isolates knowledge of pyuvm from the BFM layer. There are a lot of arguments/signals that get passed from this upper layer (from the ConfigDB) and into the lower layer. I have found that dictionaries are a great way to accomplish this task.

It would be very helpful to be able to provide default values in calls to ConfigDB().get(). I know this is not part of the UVM standard, but I believe it may fall under the "python optimization" that was done, similar to the logging changes.

[This is how Python handles this sort of thing with dictionaries](https://docs.python.org/3/library/stdtypes.html#dict.get).

This could be done by modifying the existing get() method, or adding a new get_default() or equivalent. Which would be the more "uvm" approach to this? I would be happy to implement it with some input.

**Comment by raysalemi** [comment]
We have freedom here because `ConfigDB` is not part of the UVM standard. 

I think it is appropriate to add a Pythonic solution to this problem.  

I look forward to your ideas!

**Comment by timothyscherer** [comment]
Closed by #178
